cirq/ops/common_gates.py

```diff
@@ -334,26 +334,35 @@ def __repr__(self) -> str:
 
 
 @value.value_equality
-class MeasurementGate(raw_types.Gate):
+class MeasurementGate(gate_features.MultiQubitGate):
     """A gate that measures qubits in the computational basis.
 
     The measurement gate contains a key that is used to identify results
     of measurements.
     """
 
     def __init__(self,
+                 num_qubits: int,
```

Comment:
This is still requiring the user to redundantly specify information...

Reply:
Yes.  We need to fix this.

Reply:
If we don't we are going to end up making the other protocol choices which means we will specify with_qubits or optional qubits every where and this is way worse than having it in one place.  One interface change  >> multiple changes and added complexity in large number of protocols.

Reply:
Okay. This is a relatively large breaking change. Tutorials need to be updated to use `measure` etc. I think the changes I want can be added after this PR, instead of continuing to block.

Reply:
You can blame me for this one #blamedave